demos/shape/heart/foundation.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDCShapeFoundation from '../../../packages/mdc-shape/foundation';
+
+export default class DemoHeartFoundation extends MDCShapeFoundation {
+  generatePath_(width, height, padding) {
+    return 'm ' + (width / 2) + ' ' + (height - padding)
+    +this.generateLeftBase_(width, height, padding)
+    +this.generateTopLeftCurve_(width, height, padding)
+    +this.generateTopRightCurve_(width, height, padding)
+    +this.generateRightBase_(width, height, padding);
+  }
+  generateBaseWidth_(width, padding) {
+    return ((width - (padding * 2)) / 2);
+  }
+  generateBaseFlatWidth_(width, padding) {
+    return this.generateBaseWidth_(width, padding) * 0.14;
+  }
+  generateBaseCurveWidth_(width, padding) {
+    return this.generateBaseWidth_(width, padding) * 0.52;
+  }
+  generateBaseHeight_(height, padding) {
+    return (height - (padding * 2)) * 0.67;
+  }
+  generateBaseFlatHeight_(height, padding) {
+    return this.generateBaseHeight_(height, padding) * 0.11;
+  }
+  generateBaseCurveHeight_(height, padding) {
+    return this.generateBaseHeight_(height, padding) * 0.4;
+  }
+  generateBaseCurveHeight2_(height, padding) {
+    return this.generateBaseHeight_(height, padding) * 0.67;
+  }
+  generateLeftBase_(width, height, padding) {
+    const flatWidth = this.generateBaseFlatWidth_(width, padding);
+    const flatHeight = this.generateBaseFlatHeight_(height, padding);
+    const realWidth = this.generateBaseWidth_(width, padding) - flatWidth;
+    return 'l ' + (-1 * flatWidth) + ',' + (-1 * flatHeight)
+      +'c ' + (-1 * this.generateBaseCurveWidth_(width, padding)) + ','
+      + (-1 * this.generateBaseCurveHeight_(height, padding))
+      + ' ' + (-1 * realWidth) + ',' + (-1 * this.generateBaseCurveHeight2_(height, padding))
+      + ' ' + (-1 * realWidth)
+      + ',' + (-1 * (this.generateBaseHeight_(height, padding) - flatHeight));
+  }
+  generateRightBase_(width, height, padding) {
+    const flatWidth = this.generateBaseFlatWidth_(width, padding);
+    const flatHeight = this.generateBaseFlatHeight_(height, padding);
+    const curveHeight = this.generateBaseHeight_(height, padding) - flatHeight;
+    const realWidth = this.generateBaseWidth_(width, padding) - flatWidth;
+    return 'c 0,' + (curveHeight - this.generateBaseCurveHeight2_(height, padding))
```

Comment:
This looks like a good place to use template strings in JS to make this return statement a bit simpler.